modules/at24mac.c

```diff
@@ -104,12 +109,24 @@ size_t at24mac_write( uint8_t id, uint16_t address, uint8_t *tx_data, size_t buf
             page_buf[0] = curr_addr;
             memcpy(&page_buf[1], tx_data+tx_len, bytes_to_write);
 
-            /* Write the data */
-            tx_len += xI2CMasterWrite( i2c_interface, i2c_addr, &page_buf[0] , bytes_to_write+1 );
-            tx_len -= 1; /* Remove byte address from data written size */
-            curr_addr += bytes_to_write;
+            if (i2c_take_by_chipid( id, &i2c_addr, &i2c_interface, timeout )) {
+                /* Write the data */
+                i2c_written = xI2CMasterWrite( i2c_interface, i2c_addr, &page_buf[0] , bytes_to_write+1 );
+                /* If the i2c receives a NACK when trying to write, the function returns only
+                 * the number of bytes successfully written, and we should increment only this value
+                 * in tx_len and in curr_addr */
```

Comment:
This explanation is missing some context. You should explain that the EEPROM will reply with NACKs when it is busy.